### DISCUSSION TITLE
Regarding Student Data Access Permissions

### DISCUSSION THREAD

**Discussion by cushdog** [PrairieLearn/PrairieLearn#6859] [Q&A]
Hi!

I'm a student/hobbyist, and I'm working on an application that gathers data from the grade book page and does the calculations necessary to provide an overall course grade. I sincerely apologize if this is not an appropriate platform for a question like this; I couldn't find anything in the documentation that clarified whether it's possible to fetch the JSON files corresponding to an individual student while using a student account. For some context, I have developed with PL before and know how to retrieve JSON-formatted information corresponding to individual students while logged into an account with elevated permissions, utilizing the following: `https://www.prairielearn.org/pl/api/v1/course_instances/course_instance/gradebook?private_token=example_access_token`. However, since I couldn't find any clarification in the documentation, I wanted to see if it might be possible to fetch the JSON file that only stores the specific grading information corresponding to the logged-in student account. 

Thank you for any clarification! And of course, if this question is better suited for a different forum, would anyone be able to kindly point me in the right direction? Thanks again!

**Comment by jonatanschroeder** [comment]
Hi, @cushdog, yes, this is an appropriate forum to discuss this issue. This is currently not implemented in the interface. Note that this request, to allow non-instructors to access data via API, has been requested by other users as well (see e.g., #5312 and #6525). If you are interested, we'd love to see a PR that addresses this issue.

**Reply by cushdog** [reply]
This is very informational, thank you so much! Is a student such as myself allowed to begin work on a PR?

**Reply by nwalters512** [reply]
Absolutely! PrairieLearn is open-source, so anyone can contribute.

**Reply by jonatanschroeder** [reply]
Here are some guidelines: https://github.com/PrairieLearn/PrairieLearn/blob/master/CONTRIBUTING.md
